Mage/src/main/java/mage/abilities/effects/common/ExileTargetCardCopyAndCastEffect.java

```diff
@@ -0,0 +1,59 @@
+package mage.abilities.effects.common;
+
+import mage.ApprovingObject;
+import mage.abilities.Ability;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.constants.Outcome;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.players.Player;
+
+public class ExileTargetCardCopyAndCastEffect extends OneShotEffect {
+
+    private final boolean optional;
+
+    public ExileTargetCardCopyAndCastEffect() {
```

Comment:
Maybe add a comment description here? It would be a good place to warn that text needs to be set manually. (Could also try to generate default text, though the wording would need to be adjusted more often than not so maybe not worth.)

Reply:
Yeah there are too many variations in how this effect is worded I think. I'll throw a comment in there.